src/Enum.php

```diff
@@ -2,425 +2,170 @@
-     *
-     * @internal
-     * @see \Spatie\Enum\Enum::make()
-     *
-     * @param string|null $name
-     * @param string|null $value
-     * @param int|null $index
-     */
-    public function __construct(?string $name = null, ?string $value = null, ?int $index = null)
-    {
-        if (is_null($name) && is_null($value) && is_null($index)) {
-            ['name' => $name, 'value' => $value, 'index' => $index] = $this->resolveByStaticCall();
-        }
-
-        if (is_null($name) || ! static::isValidName($name)) {
-            throw new InvalidNameException($name, static::class);
-        }
+    private static array $definitionCache = [];
 
-        if (is_null($value) || ! static::isValidValue($value)) {
-            throw new InvalidValueException($value, static::class);
-        }
+    public static function toArray(): array
+    {
+        $array = [];
 
-        if (is_null($index) || ! static::isValidIndex($index)) {
-            throw new InvalidIndexException($index, static::class);
+        foreach (static::resolveDefinition() as $definition) {
+            $array[$definition->value] = $definition->label;
         }
 
-        $this->name = $name;
-        $this->value = $value;
-        $this->index = $index;
+        return $array;
     }
 
-    public function __call(string $name, array $arguments)
+    /**
+     * @param string|int $value
+     */
+    public static function make($value): Enum
     {
-        if (static::startsWith($name, 'is')) {
-            return $this->isEqual(substr($name, 2));
-        }
-
-        if (static::isValidName($name)) {
-            return static::make($name);
-        }
-
-        throw new BadMethodCallException('Call to undefined method '.static::class.'->'.$name.'()');
+        return new static($value);
     }
 
     /**
-     * @param string $name
-     * @param mixed[] $arguments
-     *
-     * @return \Spatie\Enum\Enumerable|bool
+     * @param string|int $value
      */
-    public static function __callStatic(string $name, array $arguments)
+    public function __construct($value)
     {
-        if (static::startsWith($name, 'is')) {
-            if (! isset($arguments[0])) {
-                throw new ArgumentCountError('Calling '.static::class.'::'.$name.'() in static context requires one argument');
-            }
+        $definition = $this->findDefinition($value);
 
-            return static::make($arguments[0])->$name();
-        }
+        if ($definition === null) {
+            $enumClass = static::class;
 
-        if (static::isValidName($name) || static::isValidValue($name)) {
-            return static::make($name);
+            throw new BadMethodCallException("There's no value {$value} defined for enum {$enumClass}, consider adding it in the docblock definition.");
         }
 
-        throw new BadMethodCallException('Call to undefined method '.static::class.'::'.$name.'()');
-    }
-
-    public function __toString(): string
```

Comment:
I think that we should keep this method and return `value` - so the new renamed `value` - the one that could be the methodName, index or a custom string. So the identifier `value`.^^

This would allow to simply do `{{ $enum }}` or `trans("enums.status.{$enum}")` (Laravel).

Reply:
https://github.com/spatie/enum/pull/59/commits/b599251bfee7044c802c33f2fed73f627212d628